GNUmakefile

```diff
@@ -8,6 +8,10 @@ default: build
 build: fmtcheck
 	go install
 
+dev: fmtcheck
+	go build -o terraform-provider-vault
+	mv terraform-provider-vault ~/.terraform.d/plugins/
```

Comment:
With 386b631 this will error if `~/.terraform.d/plugins/` doesn't already exist. Is a `dev` target consistent with other Terraform providers?  I think for now we should either remove this target or leave it as `go build -o terraform-provider-vault` and leave it in the current directory to be picked up by Terraform, since `.` is a valid search path as a convenience during development 

<details>

```
[terraform-provider-vault][add-directory-structure](3)$ make dev                                                                                             
==> Checking that code complies with gofmt requirements...
go build -o terraform-provider-vault
mv terraform-provider-vault ~/.terraform.d/plugins/
mv: rename terraform-provider-vault to /Users/clint/.terraform.d/plugins/: No such file or directory
make: *** [dev] Error 1
```

</details>

Reply:
Hey, regarding this, did you see my question earlier?
>I haven't seen it in other providers, I simply did it to speed my own development workflow. Is there a different approach you're aware of that people tend to do?

Reply:
From what I saw in https://github.com/terraform-providers/ other providers just use the default make step of `build`, which just does `go install`. For development purposes I think we should leave it as is, unless `make` alone wasn't working for you? I have heard some changes may come here in the search paths for provider binaries but from what I did gather I believe if we wanted to change beyond `go install` then the right thing would be to build it in the current directory.

Reply:
This is done in https://github.com/terraform-providers/terraform-provider-vault/pull/738/commits/68eba094b3ff8a9af375c50cb0a119ab6215215b. I hadn't realized I can do essentially the same thing with just `make`.